Swerve/SwerveModules/SwerveModule.java

```diff
@@ -25,15 +26,36 @@
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import org.littletonrobotics.junction.AutoLog;
+import org.littletonrobotics.junction.Logger;
 
 public abstract class SwerveModule {
     public SwerveModuleConstants swerveModuleConstants;
     protected CANcoder angleEncoder;
     protected Rotation2d lastAngle;
-    
+
+    private final ModuleIOInputsAutoLogged inputs = new ModuleIOInputsAutoLogged();
+
     protected LinearVelocity lastVelo;
     protected Time lastVeloTime;
 
+    @AutoLog
+    public static class ModuleIOInputs {
+
+      public SwerveModuleState swerveState = new SwerveModuleState();
+      public SwerveModulePosition swervePose = new SwerveModulePosition();
+
+      public Distance driveMotorDistance = Units.Meters.zero();
+      public LinearVelocity driveVelocityMetersPerSec = Units.MetersPerSecond.zero();
+      public Current driveCurrentAmps = Units.Amps.zero();
+      public Voltage driveAppliedVolts = Units.Volts.zero();
```

Comment:
I think daniel said to remove this, please do

Reply:
something about how voltage never changes right?

Reply:
removed it